lib/torrest/api.py

```diff
@@ -82,7 +83,11 @@ class TorrestError(Exception):
 
 class Torrest(object):
```

Comment:
This class/file must be agnostic to any kodi settings. I suggest passing ssl_enabled as an input for `__init__` and if true, the Torrest api would use https protocol. Also, the port should still be configurable (like it was before). One can hava https over a different port than 443.

Reply:
Example:
```python
def __init__(self, host, port, ssl_enabled=False, session=None):
        self._base_url = "{}://{}:{}".format("https" if ssl_enabled else "http", host, port)
        self._session = session or requests
```

Reply:
Yes, you are right regarding the class that should be kodi agnostic and the port for https that can vary. I will update as per your instructions.